docs/source/geopmexporter.1.rst

```diff
@@ -0,0 +1,194 @@
+
+.. code-block:: none
+
+    geopmexporter [-h] [-v] [-t PERIOD] [-p PORT] [-i CONFIG_PATH] [--summary SUMMARY]
+
+
+Description
+-----------
+
+Prometheus exporter client that publishes telemetry read through GEOPM.  By
+default, this exports all power, energy, frequency and thermal metrics that are
+available through GEOPM on the platform.  By using the ``-i`` option, the
+exporter may be configured to publish any set of available signals.
+
+The metrics are sampled periodically through the GEOPM PlatformIO batch
+interface.  The period of these samples defaults to 100 ms, but the user may
+override this with the ``-t`` option.  The values measured are summarized when
+the Prometheus server scrapes the data published by the exporter.  These data
+are published on port 8000 by default, but the user may override this with the
+``-p`` option.
+
+
+Options
+~~~~~~~
+
+-h, --help  .. _help option:
+
+    Print help message and exit
+
+-v, --version  .. _version option:
+
+    Print version and exit
+
+-t, --period  .. _period PERIOD option:
+
+    Sample period for fast loop in seconds. Default: 0.1
+
+-p, --port  .. _port PORT option:
+
+    Port to publish Prometheus metrics. Default: 8000
+
+-i, --signal-config  .. _signalconfig CONFIG_PATH option:
+
+    Signal configuration file path, specify "-" to use standard input.  The
+    format of the configuration file is described below.
+
+--summary  .. _summary SUMMARY option:
+
+    Summary method, one of "geopm", or "prometheus". Default: geopm
+
+
+Configuration File
+------------------
+
+The input file is a list of read requests each on its own line.  A request for
+reading is made up of three strings separated by white space. The first string
+is the signal name, the second string is the domain name, and the third string
+is the domain index.  Provide the "``*``" character as the third string to
+request all domains available on the system.
+
+
+GEOPM Summary Method
+--------------------
+
+There are two options available for summarizing the statistics for the
+Prometheus exporter.  The default option is to use the GEOPM ``StatsCollector``
+interface to summarize sampled values.  This is a high performance option that
+provides fine grained summary statistics through a Prometheus Gauge metric
+interface:
+
+count:
+   Number of samples recorded since last Prometheus server scrape
+
+first:
+   First value sampled after last Prometheus server scrape
+
+last:
+   Most recently sampled value prior to Prometheus server scrape
+
+min:
+   Smallest sampled value since last Prometheus server scrape
+
+max:
+   Largest sampled value since last Prometheus server scrape
+
+mean:
+   Sum of all samples divided by the number of samples since last Prometheus
+   server scrape
+
+std:
+   Sample standard deviation represented by the following sum over samples:
+
+       ``sqrt(sum((x_i - m) ^ 2) / (n - 1))``
+
+   where ``m`` is the mean, ``x_i`` are the samples, and ``n`` is the number of
+   samples
```

Comment:
Does the number of samples keep increasing forever (util the StatsCollector object is destroyed)?

Reply:
Note the memory requirements for the collector do not change as more samples are added.